Features/bootstrap/MainContext.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+use QATools\BehatExtension\Context\QAToolsContext;
+
+class MainContext extends QAToolsContext
+{
+
+	/**
+	 * @When /^the user enters (.+?) and (.+?)$/
+	 */
+	public function fillForm($username, $password)
+	{
+		$this->page->fillForm($username, $password);
+	}
+
+	/**
+	 * @Then /^the login should fail$/
+	 */
+	public function loginFail()
+	{
+		$this->page = $this->page->loginFailure();
```

Comment:
In that case always combine submit and filling the form? I always divided that but it could also be one step after all.